errors.go

```diff
@@ -7,7 +7,7 @@ import (
 
 var (
 	ErrInvalidKey                = errors.New("key is invalid")
-	ErrInvalidKeyType            = errors.New("key is of invalid type")
+	ErrInvalidKeyType            = errors.New("key is of invalid go type")
```

Comment:
I would prefer to keep the error string the same of backwards compatibility. I think if we include the additional details, it is fairly obvious what this error is. If you want to stress the point from a documentation point of view, we can add a godoc here.

Reply:
Yeah okay that makes sense I agree, since that error is only used in the context with the new information that should suffice